### DISCUSSION TITLE
Is there any plan to implement the useTree hook?

### DISCUSSION THREAD

**Discussion by loicplaire** [adobe/react-spectrum#6945] [Q&A] [answered]
Hi!

We built our components library using React Aria hooks. On our roadmap, we will soon have to rebuild our Tree component. I know the Tree component has recently been released and I wondered if there was any plan to provide a `useTree` hook.

I noticed there is currently [an empty hook](https://github.com/adobe/react-spectrum/blob/main/packages/%40react-aria/tree/src/useTree.ts) in the repository but I couldn't find any mention of it anywhere else.

This is the UX of the component that we want to build.

<img width="795" alt="Screenshot of the existing Tree component" src="https://github.com/user-attachments/assets/f817295e-e4e3-4bbd-a8cd-71fea89d3c8f">

If there no `useTree` hook built plan soon, what would be your recommendation in terms of hooks to use? 

Building a component similar to [the React Aria Components Tree](https://github.com/adobe/react-spectrum/blob/main/packages/react-aria-components/src/Tree.tsx) or getting some inspiration from [this example that was shared a few years ago](https://github.com/adobe/react-spectrum/discussions/3698#discussioncomment-4015913)?

Thanks for your help!

**Comment by reidbarber** [comment, marked as answer]
The hook for this is [useTreeGridList](https://github.com/adobe/react-spectrum/blob/dcf312bc21ac08e9202e110f10572d4df47784b1/packages/%40react-aria/tree/src/useTreeGridList.ts). You can reference the RAC Tree code you linked to see how to use it. Docs are coming later, and as it in still in an alpha state, naming/API could change.
